src/features/runtime-checks.js

```diff
@@ -573,4 +576,211 @@ export default class RuntimeChecks extends ContentFeature {
+                return Object.prototype.hasOwnProperty.call(this.#data, id) ? this.#data[id] : undefined
+            }
+
+            removeItem (id) {
+                return delete this.#data[id]
+            }
+
+            clear () {
+                this.#data = {}
+            }
+
+            key (n) {
+                const keys = Object.keys(this.#data)
+                return keys[n]
+            }
+
+            get length () {
+                return Object.keys(this.#data).length
+            }
+        }
+
+        const storage = new MemoryStorage()
+        this.overrideStorage(config, key, storage)
+    }
+
+    overloadStorageWithSession (key, config) {
+        const storage = globalThis.sessionStorage
+        this.overrideStorage(config, key, storage)
+    }
+
+    overrideStorage (config, key, storage) {
+        const originalStorage = globalThis[key]
+        defineProperty(globalThis, key, {
+            get () {
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    return storage
+                }
+                return originalStorage
+            }
+        })
+    }
+
+    /**
+     * @typedef {object} Sizing
+     * @property {number} height
+     * @property {number} width
+     */
+
+    /**
+     * Overloads the provided key with the closest breakpoint size
+     * @param {Sizing[]} breakpoints
+     * @param {Sizing} screenSize
+     * @param {string} key
+     * @param {number} [offset]
+     */
+    overloadScreenSizes (config, breakpoints, screenSize, key, offset = 0) {
+        const closest = findClosestBreakpoint(breakpoints, screenSize)
+        if (!closest) {
+            return
+        }
+        let returnVal = null
+        /** @type {object} */
+        let scope = globalThis
+        let overrideKey = key
+        let receiver
+        switch (key) {
+        case 'innerHeight':
+        case 'outerHeight':
+            returnVal = closest.height - offset
+            break
+        case 'innerWidth':
+        case 'outerWidth':
+            returnVal = closest.width - offset
+            break
+        case 'Screen.prototype.height':
+            scope = Screen.prototype
+            overrideKey = 'height'
+            returnVal = closest.height - offset
+            receiver = globalThis.screen
+            break
+        case 'Screen.prototype.width':
+            scope = Screen.prototype
+            overrideKey = 'width'
+            returnVal = closest.width - offset
+            receiver = globalThis.screen
+            break
+        }
+        const defaultVal = Reflect.get(scope, overrideKey, receiver)
+        defineProperty(scope, overrideKey, {
+            get () {
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    return returnVal
+                }
+                return defaultVal
+            }
+        })
+    }
+}
+
+function findClosestBreakpoint (breakpoints, screenSize) {
```

Comment:
is there a simple unit-test we can add to this? seems like low-hanging fruit and would help to explain the math :)